src/Grpc.AspNetCore.Server/Internal/DefaultGrpcInterceptorActivator.cs

```diff
@@ -0,0 +1,72 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using Grpc.Core.Interceptors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal class DefaultGrpcInterceptorActivator<TInterceptor> : IGrpcInterceptorActivator<TInterceptor> where TInterceptor : Interceptor
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private bool? _created;
+
+        public DefaultGrpcInterceptorActivator(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public Interceptor Create(params object[] args)
+        {
+            Debug.Assert(!_created.HasValue, "Grpc interceptor activator must not be reused.");
+
+            _created = false;
+
+            if (args.Length == 0)
+            {
+                var globalInterceptor = _serviceProvider.GetService<TInterceptor>();
+                if (globalInterceptor != null)
+                {
+                    return globalInterceptor;
+                }
+            }
+
+            var interceptor = ActivatorUtilities.CreateInstance<TInterceptor>(_serviceProvider, args);
```

Comment:
Hmm, this uses reflection every time. We can figure out perf here, and add some micro-benchmarks, in a follow up PR